src/neptune/metadata_containers/run.py

```diff
@@ -451,6 +461,43 @@ def _write_initial_attributes(self):
             # upload default sources ONLY if creating a new run
             upload_source_code(source_files=self._source_files, run=self)
 
+        if self._dependencies:
+            self._track_dependencies()
+
+    def _track_dependencies(self) -> None:
```

Comment:
Let's extract this logic to separate class that would be easier to test. Here you could only check that it calls some class methods correctly and in another test suite ensure that it works for `infer`/path-specific.